dace/codegen/targets/intel_fpga.py

```diff
@@ -519,17 +532,14 @@ def generate_module(self, sdfg, state, name, subgraph, parameters,
         is_autorun = len(kernel_args_opencl) == 0
 
         # create a unique module name to prevent name clashes
-        module_function_name = "module_" + name + "_" + str(sdfg.sdfg_id)
-
-        # The official limit suggested by Intel is 61. However, the compiler
+        module_function_name = "mod_" + str(sdfg.sdfg_id) + "_" + name
+        # The official limit suggested by Intel for module name is 61. However, the compiler
         # can also append text to the module. Longest seen so far is
         # "_cra_slave_inst", which is 15 characters, so we restrict to
         # 61 - 15 = 46, and round down to 42 to be conservative.
-        if len(module_function_name) > 42:
-            raise NameTooLongError(
-                "Due to a bug in the Intel FPGA OpenCL compiler, "
-                "kernel names cannot be longer than 42 characters:\n\t{}".
-                format(module_function_name))
+        # Therefore we cut down names longer than that
+        module_function_name = module_function_name[0:42]
```

Comment:
Then we need a check for collisions!

Reply:
isn't the `sdfg.id` unique?

Reply:
Ah, are you appending this _after_ you cut this off? Then you're right

Reply:
Yep It is prepended ` module_function_name = "mod_" + str(sdfg.sdfg_id) + "_" + name`

Reply:
Ok, great!